File: app_frontend/src/models/Apartment.js
export default class Apartment {
  constructor({ apartmentId, name, owner, address, description, city, country, price }) {
    this.apartmentId = apartmentId;
    this.name = name;
    this.owner = owner;
    this.address = address;
    this.description = description;
    this.city = city;
    this.country = country;
    this.price = price;
  }

  displayInfo() {
    return `${this.name} at ${this.address}, ${this.city}, ${this.country} — $${this.price}`;
  }

  ownerName() {
    return this.owner?.username || "Unknown Owner";
  }
}

File: FlatM8/src/main/java/FlatM8/model/Apartment.java
package FlatM8.model;

import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Apartment {
    private UUID apartmentId;
    private String name;
    @ManyToOne
    @JoinColumn(name = "owner_id")
    private User owner;
    private String address;
    private String description;

}
